src/plots/gl2d/scene2d.js

```diff
@@ -522,6 +522,21 @@ proto.draw = function() {
                             glplot.pixelRatio
                 ];
 
+                // this needs to happen before the next block that deletes traceCoord data
+                // also it's important to copy, otherwise data is lost by the time event data is read
+                this.graphDiv.emit('plotly_hover', {
+                    points: [{
+                        trace: nextSelection.trace,
+                        dataCoord: nextSelection.dataCoord.slice(),
+                        traceCoord: nextSelection.traceCoord.slice(),
+                        textLabel: nextSelection.textLabel,
+                        color: nextSelection.color,
+                        name: nextSelection.name,
+                        hoverinfo: nextSelection.hoverinfo,
+                        screenCoord: nextSelection.screenCoord.slice()
```

Comment:
> Do you think it's possible to make it work just by invoking it from within scene2d

It's definitively possible. That's how I got mapbox hover labels to work - see [here](https://github.com/plotly/plotly.js/blob/fce36fa46bd53fa5ebe4e4892479258040783409/src/plots/mapbox/mapbox.js#L164).

But, I would call that outside the scope of this PR. 

I think you already have all the info required to produce the same event data object within `scene2d`.